compat/mingw.c

```diff
@@ -508,6 +508,9 @@ int mingw_unlink(const char *pathname)
 	if (xutftowcs_long_path(wpathname, pathname) < 0)
 		return -1;
 
+	if (DeleteFileW(wpathname))
```

Comment:
For my understanding. Why is this not using `_wunlink` like below?  What is this difference between `DeleteFileW` and `_wunlink`?  Also should it be trying the `_wrdir` like below?  Should lines 517-527 be pulled out into a method that can be called here and in the `while` loop?

Reply:
This is a probabilistic effort.  Most of the time your worktree is 0666 and DeleteFileW() will just succeed and be done.  And most of the time you're trying to delete files rather than
directories.  (That second point is speculation, but again I'm optimizing for the happy-path.)

_wunlink() is a routine in the MSFT CRT that just calls DeleteFileW() and then their custom (and private) GetLastError()-2-errno function if there's an error.  So I'm just short-cutting
for the happy-path.  But if that fails for any reason, I let the existing logic (both the
function calls and the loop) handle it so there's no semantic changes for the error
handling.  (Some of this logic is a bit questionable, but that's outside of the scope
of what I wanted to do in this fix.)